[CLD-215]: fix(environment): remove migrated_environment.go

graham-chainlink opened the pull request:
Remove type alias in migrated_environment.go now that dependent repository like CLD are migrated off from using the alias

- removed [migrated_environment.go
](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/migrated_environment.go)

Due to an issue documented [here](https://github.com/golang/go/issues/69179), this occurs when adding a new import for chainlink-deployments-framework/deployments, i need to set the goindex=0 for godebug. There was a line setting it already but it didn not apply correctly. Found some historical PR related to this - [here](https://github.com/smartcontractkit/chainlink/pull/16393) and [here](https://github.com/smartcontractkit/chainlink/pull/16362)

Slack thread: https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1747107991420549

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-215

**Lint issues are expected as i touched a range of files and those are existing lint issues that were picked up, i am ignoring them using `allow-lint-issues`**

jkongie reviewed (commented):
Reviewed all the import changes, and they look good.

I'm still a bit unsure about the change to the github workflow, so will need someone else to double check that.

graham-chainlink commented:
> Reviewed all the import changes, and they look good.
> 
> I'm still a bit unsure about the change to the github workflow, so will need someone else to double check that.

thanks @jkongie , i have a running [thread here](https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1747107991420549), hopefully we get some core people or erik to have a quick look at this PR.

graham-chainlink committed:
fix(environment): remove alias

Remove type alias to CLDF repository now that dependent repository like CLD are migrated off from using the alias

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-215

graham-chainlink committed:
ci: set goindex=0

Due to an issue documented [here](golang/go#69179), this occurs when adding a new import for chainlink-deployments-framework/deployments, i need to set the goindex=0 for godebug. There was a line setting it already but it didn not apply correctly.

More context: https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1747128640637199?thread_ts=1747107991.420549&cid=C0117GGJB6Y
